lib/Dialect/TritonGPU/Transforms/Pipeliner/PipeliningUtility.cpp

```diff
@@ -73,6 +73,19 @@ Operation *mlir::triton::predicateOp(RewriterBase &rewriter, Operation *op,
     expectOp.getPredMutable().assign(mask);
     return op;
   }
+  if (auto storeOp = dyn_cast<tt::StoreOp>(op)) {
+    rewriter.setInsertionPoint(storeOp);
+    // create conditional store
+    auto ifOp = rewriter.create<scf::IfOp>(storeOp.getLoc(), pred, false);
+    auto *b = ifOp.thenBlock();
+    op->moveBefore(b, b->begin());
+    return ifOp;
+  }
+  if (auto dotOp = dyn_cast<ttng::WarpGroupDotOp>(op)) {
```

Comment:
Don't think we need this to be supported? Can you drop it?

Reply:
I can remove it but will fail one of the generic tests.

@ThomasRaoux is this op really side-effecting?

Reply:
it is side-effecting as it reads shared memory. Why would it fail a test? Wouldn't the pipeliner just fail?

Reply:
Would this be ok to speculatively execute?

If this op is not handled it will trigger the assertion below. I could however return nullptr and the pass would fail.

Reply:
no it wouldn't always be correct to speculatively execute it. Returning nullptr would work although that means now this would just fail the pipelining instead of giving an error? I guess this is probably okay. The alternative is to move the wgmma case to the hopper test file

Reply:
I'll move this to hopper since support can be added correctly there.

I've also created a PR upstream to gracefully handle when predicateFn returns nullptr in all cases (it was only missing from the prologue call). 
https://github.com/llvm/llvm-project/pull/107442